pulsar-transformations/src/test/java/com/datastax/oss/pulsar/functions/transforms/jstl/JstEvaluatorTest.java

```diff
@@ -46,6 +48,47 @@ void testFunctions(String expression, TransformContext context, Object expectedV
         new JstlEvaluator<>(String.format("${%s}", expression), Object.class).evaluate(context));
   }
 
+  @Test
+  void testNowFunction() {
+    Record<GenericObject> primitiveStringRecord =
+        new Utils.TestRecord<>(
+            Schema.STRING,
+            AutoConsumeSchema.wrapPrimitiveObject("test-message", SchemaType.STRING, new byte[] {}),
+            "header-key");
+    TransformContext primitiveStringContext =
+        new TransformContext(
+            new Utils.TestContext(primitiveStringRecord, new HashMap<>()),
+            primitiveStringRecord.getValue().getNativeObject());
+
+    long expectedNow = Instant.now().toEpochMilli();
+    long actualNow =
+        new JstlEvaluator<Long>("${fn:now()}", long.class).evaluate(primitiveStringContext);
+
+    assertTrue(actualNow >= expectedNow);
```

Comment:
Maybe we could have the possibility to pass a custom Clock to the class under test.
This way it's easier to control time.

Reply:
JUEL only supports binding of static method, so I have to do it the suboptimal way by introducing a clock on the static class and modify it by tests (as opposed to dependency injection)